### PR TITLE
refactor: add comment for getAuthenticationState

### PR DESCRIPTION
**Pull request by robrap**

### CODE REVIEW

File: src/AuthenticatedAPIClient/authInterface.js [hunk 1]
```diff
@@ -29,6 +29,12 @@ export default function applyAuthInterface(httpClient, authConfig) {
     httpClient.refreshAccessTokenEndpoint,
   ];
 
+  /**
+   * IMPORTANT: Do not use this function outside of frontend-auth.
+   * In certain cases, the token might expire between when frontend-auth
+   * authenticated and this call is made. Instead, use the token returned
+   * by the callback.
+   */
```
Comment (douglashall) [inline src/AuthenticatedAPIClient/authInterface.js]: To me this statement is confusing and not entirely accurate. I wouldn't add this.

Comment (robrap) [reply src/AuthenticatedAPIClient/authInterface.js]: Can you help me make it less confusing and more accurate?  What about?:
```
/**
   * IMPORTANT: This is a private function.  From outside this library,  
   * only use the token returned by the callback to guarantee you are
   * using the same token used for authentication.
   */
```

Comment (douglashall) [reply src/AuthenticatedAPIClient/authInterface.js]: I think the best course of action here is to have no comment.

Comment (robrap) [reply src/AuthenticatedAPIClient/authInterface.js]: Let me explain why I think this comment is important.  You found something that is not obvious, and I don't believe the code alone explains what you found, so I want to capture it for future folk like me.

If you still believe it is better to not have a comment, I'd like to better understand why.

Additionally, there are still many outside callers and we'll need to inform them about this bug if you haven't already: https://github.com/search?q=org%3Aedx+getAuthenticationState&type=Code

**Pull request merged**